docs-chef-io/content/server/upgrade_server_ha_v2.md

```diff
@@ -163,13 +163,193 @@ Upgrading from Chef Backend 1.x to Chef Backend 2.x requires full cluster downti
+
+#### Identify the Front-end Chef Infra Server Versions
+
+Find the Chef Infra Server version on each of your front-end nodes, either from the `/opt/opscode/version-manifest` file or by using the command:
+
+```ruby
+chef-server-ctl version
+```
+
+#### Identify the Front-end Elasticsearch Versions
+
+Find the version of Elasticsearch that created the indexes with the command:
+
+```bash
+curl -XGET localhost:9200/chef/_settings/index.version.created*?pretty\&human
+```
+
+#### Upgrade Strategy
+
+The Elasticsearch upgrade uses a rolling strategy, but the Elasticsearch version used to create the indexes determines if your upgrade is automatic or requires downtime.
+
+Chef Infra Servers with Elasticsearch 5.4.1 or lower require downtime. Systems with Elasticsearch 5.6.16 or higher are automatic and do not require downtime.
+
+A migration within a major version, for example, from 2.0.1 to 2.3.16, is always Automatic.
+
+| Running Backend Version | Upgrade Target | Elasticsearch Version | Upgrade Strategy |
+|---------|--------|------|------|
+| 2.3.16 | 3.0 | `"created_string" : "5.6.16"` | Automatic |
+| 2.2.0 | 3.0 | `"created_string" : "5.6.16"` | Automatic |
+| 2.1.0 | 3.0 | `"created_string" : "5.6.16"` | Automatic |
+| 2.0.30 | 2.3.16 | `"created_string" : "5.4.1"` | Downtime |
+| 1.4.6 | 2.3.16 | `"created_string" : "2.3.1"` | Downtime |
+| 1.4.6 | 2.3.16 | `"created_string" :` Any  | Downtime |
+| 1.3.2 | 2.3.16 | `"created_string" :` Any  | Downtime |
+| 1.2.5 | 2.3.16 | `"created_string" :` Any  | Downtime |
+| 1.1.2 | 2.3.16 | `"created_string" :` Any | Downtime |
+| 1.0.2 | 2.3.16 | `"created_string" :` Any | Downtime |
+
+#### Index Treatment
+
+The the version of Elasticsearch used to create the indexes determines the how the upgrade treats the indexes.
+
+* Indexes created by Elasticsearch 2.3.0 or earlier _and then_ upgraded to version 2.3.1 need to be deleted and reindexed after the upgrade.
+* Indexes crated by Elasticsearch 2.3.1 or later will automatically re-index.
+
+| Original Version | Upgraded Version| Elasticsearch Version | Upgraded Version Index Treatment |
+|---------|--------|--------|--------|
+| 2.3.16 | 3.0 | `"created_string" : "5.6.16"` | Automatic reindex |
+| 2.2.0 | 3.0 | `"created_string" : "5.6.16"` | Automatic reindex |
+| 2.1.0 | 3.0 | `"created_string" : "5.6.16"` | Automatic reindex |
+| 2.0.30 | 2.3.16 | `"created_string" : "5.4.1"` | Delete and reindex |
+| 1.4.6 | 2.3.16 | `"created_string" : "2.3.1"` | Delete and reindex |
+| 1.4.6 | 2.3.16 | `"created_string" :` Any  | Delete and reindex |
+| 1.3.2 | 2.3.16 | `"created_string" :` Any  | Delete and reindex |
+| 1.2.5 | 2.3.16 | `"created_string" :` Any  | Delete and reindex |
+| 1.1.2 | 2.3.16 | `"created_string" :` Any | Delete and reindex |
+| 1.0.2 | 2.3.16 | `"created_string" :` Any | Delete and reindex |
+
+### Upgrade Front-end Servers with Elasticsearch 1.7-5.4.1
```

Comment:
This upgrade is specifically for backend, so we would not necessarily be upgrading the frontends. Also frontends would never have elasticsearch installed.